### ISSUE TITLE
Number of estimators.

### ISSUE THREAD

**Issue by JinyuanSun** [LiamT01/CTR#3]
https://github.com/LiamT01/CTR/blob/917f010b7cd8c73fe9b5b15d03ea19d224b08360/model_config.py#L10

How much will the number of estimators affect the accuracy of the final model?
Setting `n_estimators` to 2000 will cost about 2~3 days to train on my server with 2 Gold 6248R CPUs (96 threads in total). 
I changed it to 200, metrics at the 3rd epoch are: 
```
rmse=1.0029, mse=1.0058, pearsonr=0.8368
```
when n_estimators = 400,
```
rmse=0.9730, mse=0.9468, pearsonr=0.8440
```
Improvement seems marginal and close to the correlation reported in the paper.

**Comment by LiamT01** [comment]
The number of estimators would not have a significant impact on the accuracy. At the time of experiments, we did not overly tune the hyperparameters because the improvements were marginal, as you pointed out.

However, even with `n_estimators=2000`, training took only 10 minutes using the M2 chip and no more than 30 minutes on our server using one Intel Xeon E5-2670 CPU. Please check if other issues might have affected the training speed. Also, check if the environment has been correctly set up with the latest packages.

**Comment by JinyuanSun** [comment]
Thanks for your information! I did run a lower version. However, after creating a new env:
```
# packages in environment at /home/jsun/anaconda3/envs/ctr:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2022.9.24            ha878542_0    conda-forge
joblib                    1.2.0                    pypi_0    pypi
ld_impl_linux-64          2.39                 hcc3a1bd_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 12.2.0              h65d4601_19    conda-forge
libgomp                   12.2.0              h65d4601_19    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libsqlite                 3.40.0               h753d276_0    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libzlib                   1.2.13               h166bdaf_4    conda-forge
ncurses                   6.3                  h27087fc_1    conda-forge
numpy                     1.23.5                   pypi_0    pypi
openssl                   3.0.7                h0b41bf4_1    conda-forge
pandas                    1.5.1                    pypi_0    pypi
pip                       22.3.1             pyhd8ed1ab_0    conda-forge
python                    3.10.8          h4a9ceb5_0_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
pytz                      2022.6                   pypi_0    pypi
readline                  8.1.2                h0f457ee_0    conda-forge
scikit-learn              1.1.3                    pypi_0    pypi
scipy                     1.9.3                    pypi_0    pypi
setuptools                65.5.1             pyhd8ed1ab_0    conda-forge
six                       1.16.0                   pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tk                        8.6.12               h27826a3_0    conda-forge
tqdm                      4.64.1                   pypi_0    pypi
tzdata                    2022g                h191b570_0    conda-forge
wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
xgboost                   1.7.1                    pypi_0    pypi
xz                        5.2.6                h166bdaf_0    conda-forge
```
Training speed did not improve.  
But thanks to the released trained model. I don't really have to fix this right now.

**Comment by LiamT01** [comment]
Thanks for your feedback that helped improve this work!
